Automattic/wp-calypso
# Move SitePreview from Layout to sections that use it

Moves the `<SitePreview />` component, one that renders a site preview modal when Redux state in `state.ui.preview` says so, from the global `<Layout />` component to the two specific sections that display it: Pages and Posts. That should make the webpack entrypoint smaller.

In addition, I'm moving the `blocks/post-item` component to `my-sites/post-type-list/post-item`, because the `<PostTypeList />` is its only user. It's a sub-component rather than a standalone component.

**How to test:**
In the Posts and Pages view (the Calypso ones), select "View" from the ellipsis menu. Verify that the `SitePreview` modal is displayed and that it doesn't show any placeholder artifacts while loading (that's why I added `placeholder={ null }` in the last commit).
